### PR TITLE
feat: Adds `-Repo` command to Helm

### PR DESCRIPTION
**Pull request by ElvenSpellmaker**
fix: Fixes build false passing by handling errors
 * Gracefully handles Pester errors by storing the exit code in a file, running the next code coverage task, and then finally running a task which will exit based on the exit code in the file. This should make the build fail now, while still running the HTML Code Coverage generation task.
 * Fixes the YAMLLint test, leaving TF deliberately failing to see if the build now fails as expected.
 * Ensures the build runs the root `tests` task and not the individual
   ones.

feat: Adds `repo` command to Helm
 * Adds a `-Repo` command for adding repos to Helm, this means it's no
   longer required to use a custom command
 * The new `-Repo` command doesn't log into a Cloud Provider
   whereas `-Custom` still does (to maintian bc)
 * Fixes the Helm tests which weren't working properly (although more
   tests should be added in future as currently there aren't exhaustive
   tests there)
 * Fixes the Terraform Tests since the change to workspace command
 * Moves not supplying a workspace to select to an Error, from Warning

fix: Fix 'missing' files by using sed
 * Fix the error 'File '/app/src/*' does not exist (any more).' by using
   sed to replace the TaskCTL Docker context bind path with the path to
   the files on the host agent.

NOTE: More Helm tests should be added in future to exhaustively test it

### CODE REVIEW

**Comment by ElvenSpellmaker** [comment]
Woot, works as intended: https://amido-dev.visualstudio.com/Amido-Stacks/_build/results?buildId=29926&view=logs&jobId=527ac849-7c2e-5ca3-43fc-79206f5491f8

**Comment by ElvenSpellmaker** [comment]
image

The `sed` command works 🙂

**Review by russellseymour — APPROVED** [review]
Thanks @ElvenSpellmaker , error handling keeps coming back :-)

**Comment by ElvenSpellmaker** [comment]
@russellseymour It always does 🙂

File: src/modules/AmidoBuild/exported/Invoke-Terraform.Tests.ps1 [hunk 1]
```diff
@@ -23,7 +23,7 @@ Describe "Invoke-Terraform" {
         Mock -Command Find-Command -MockWith { return "terraform" }
 
         Mock -CommandName Write-Error -MockWith { }
-        Mock -CommandName Write-Warning -MockWith { }
+        # Mock -CommandName Write-Warning -MockWith { }
```
Comment (ElvenSpellmaker) [inline src/modules/AmidoBuild/exported/Invoke-Terraform.Tests.ps1:26]: Why?

File: src/modules/AmidoBuild/exported/Invoke-Helm.ps1 [hunk 2]
```diff
@@ -76,34 +83,69 @@ function Invoke-Helm() {
...
+            # Check that some arguments have been set
+            $checkParams = @("provider", "target", "identifier", "namespace", "releasename","namespace")
+        }
 
+        "repo" {
+            $checkParams = @("repositoryName", "repositoryUrl")
+        }
+
+        "custom" {
+            $checkParams = @("arguments")
```
Comment (ElvenSpellmaker) [inline src/modules/AmidoBuild/exported/Invoke-Helm.ps1:116]: Custom currently has to log in, so required should also include "provider", "target", "identifier" too, and both install and custom need the "k8sauthrequired"

File: src/modules/AmidoBuild/exported/Invoke-Helm.ps1 [hunk 3]
```diff
@@ -76,34 +83,69 @@ function Invoke-Helm() {
...
-      }
+    }
 
     # if there are missing parameters throw an error
     if ($missing.length -gt 0) {
         Write-Error -Message ("Required parameters are missing: {0}" -f ($missing -join ", "))
         exit 1
+    }
+
+    $login =  {
```
Comment (ElvenSpellmaker) [inline src/modules/AmidoBuild/exported/Invoke-Helm.ps1:134]: lint

**Comment by ElvenSpellmaker** [comment]
Just left myself some notes for a "fix" PR after

**Pull request merged**